src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -238,8 +255,7 @@ DirectStepI::revImp (
 
     bool srcRedeems;
     IOUAmount maxSrcToDst;
-    std::tie (maxSrcToDst, srcRedeems) =
-        maxFlow (sb, src_, dst_, currency_);
+    std::tie (maxSrcToDst, srcRedeems) = maxFlow (sb, out);
```

Comment:
I was thinking about this as ignoring the trust line *limits*, but I guess the result is the same.  That's because the current offer crossing implementation ignores trust line limits (by design).

I'm guessing your concern is that `srcToDst` might be larger than the passed in `out` value.  If that's the case then, yeah, I have a problem to fix.  I'll need to think about that fix for a bit, though.  I'm trying to limit the damage that offer crossing does to your code.  If I'm not careful I might end rewriting `revImp` and `fwdImp`, which I'd very much like to avoid.

I think I'll start by writing a unit test that makes `srcToDst` be larger than `out`.  That will at least give me an example to work against.  Interestingly, none of the current unit tests exercise that case.